ksqldb-engine/src/main/java/io/confluent/ksql/topic/TopicCreateInjector.java

```diff
@@ -206,6 +215,10 @@ private TopicProperties createTopic(
 
     final Map<String, Object> config = new HashMap<>();
     config.put(TopicConfig.CLEANUP_POLICY_CONFIG, topicCleanUpPolicy);
+    // This is to make sure WINDOW RETENTION config is honored over WITH RETENTION_MS config
```

Comment:
shouldn't we chose the greater of the two? `WINDOW RETENTION` can be less than the topic retention (I think) it just can't be greater than

Reply:
Users might be used to the `WINDOW RETENTION` config which gets applied to the topic retention - that was the reason I chose to leave it as is. But what you said makes sense. We could choose the greater of the two.
cc @colinhicks

Reply:
Choosing the greater value sounds good to me, @bvarghese1

Reply:
did we make this change?

Reply:
No not yet. Still working on this PR. I think I need to also re-evaluate some of the tests now that I understand the cleanup.policy settings impact on retention.

Reply:
Made the change to pick the greater of the two(i.e. retention config passed via WITH and retention config specified in the WINDOW config).